packages/docprovider-extension/src/filebrowser.ts

```diff
@@ -131,10 +155,11 @@ export const ynotebook: JupyterFrontEndPlugin<void> = {
         disableDocumentWideUndoRedo
       });
     };
-    drive.sharedModelFactory.registerDocumentFactory(
+    contentProvider.sharedModelFactory.registerDocumentFactory(
       'notebook',
       yNotebookFactory
     );
+    notebookFactory.contentProviderId = 'rtc';
```

Comment:
This seems to indicate that multiple extensions could try to set their content provider on the default notebook and editor factories? 

What is the expected behavior if two extensions try to do this?

Reply:
> What is the expected behavior if two extensions try to do this?

Currently the last one wins. Alternative, we could log a warning or reject a second use of setter.

Reply:
> Currently the last one wins. Alternative, we could log a warning or reject a second use of setter.

Maybe a hard failure would be preferable for end users, so it's easier to debug and report?

Reply:
Make sense. Done in https://github.com/jupyterlab/jupyterlab/pull/17092/commits/1173ca7ec995df071142bcb1170781a0897be246.